### ISSUE TITLE
Bug in master when showing tags for a container?

### ISSUE THREAD

**Issue by diranged** [SUSE/Portus#822]
After upgrading to the latest [master](https://github.com/SUSE/Portus/commit/77c7b1582dd1689457a78677f3659088c1e87922), we are seeing tags displayed like this:

![image](https://cloud.githubusercontent.com/assets/768067/14533877/c38671da-021b-11e6-9cde-94a97781f477.png)

**Comment by mssola** [comment]
This was added quite recently in commit 28dae7f3fb233dbeb9290c30671f829c1b85edbf. From this I can see two different issues actually :sweat_smile: :

1. The crono task does not add the digest of the tag when it creates them (or at least from the `Author` on your screenshot I can see that crono was the one creating the tag).
2. The groupping of tags works in this weird way when the digest is empty. Tags with empty digests should be considered different from each other.

**Comment by mssola** [comment]
@diranged Now your issue should be fixed. Also you may want to update the tags that were added by crono without filling the manifest digest. You can do that with the rake task introduced in #830:

    $ rake portus:update_tags

**Comment by diranged** [comment]
So I updated to the latest master and I tried to run that command:
```
[root@ops-registry-useast1:~:130]# docker exec -ti portus-crono /bin/bash
root@958c5be0dc7e:/portus# rake portus:update_tags
WARNING: Nokogiri was built against LibXML version 2.9.1, but has dynamically loaded 2.8.0
[Mailer config] Host:     portus.corp.nextdoor.com
[Mailer config] Protocol: https://
fatal: Not a git repository (or any of the parent directories): .git
This rake task may take a while depending on how many images have been stored
in your private registry. If you are running this in production it's
recommended that the registry is running in "readonly" mode, so there are no
race conditions with concurrent accesses.

Are you sure that you want to proceed with this ? (y/N) Y
There are no tags to be updated.
root@958c5be0dc7e:/portus# 
```

So good, right? Not so good ..


```
Apr 19 16:18:15.121256 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]: Started GET "/repositories/60" for 162.245.21.10 at 2016-04-19 16:18:15 +0000
Apr 19 16:18:15.122167 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]: Processing by RepositoriesController#show as HTML
Apr 19 16:18:15.122204 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]:   Parameters: {"id"=>"60"}
Apr 19 16:18:15.135779 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]:   Rendered repositories/show.html.slim within layouts/application (5.6ms)
Apr 19 16:18:15.135779 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]: Completed 500 Internal Server Error in 14ms (ActiveRecord: 6.5ms)
Apr 19 16:18:15.136928 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]: 
Apr 19 16:18:15.136928 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]: ActionView::Template::Error (undefined method `empty?' for nil:NilClass):
Apr 19 16:18:15.136928 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]:     28:             th Pushed at
Apr 19 16:18:15.136928 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]:     29:         tbody      
Apr 19 16:18:15.136928 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]:     30:           - @tags.each do |tag|
Apr 19 16:18:15.136928 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]:     31:             - if tag.first.digest.empty?
Apr 19 16:18:15.137070 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]:     32:               - tag.each do |t|
Apr 19 16:18:15.137070 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]:     33:                 tr         
Apr 19 16:18:15.137070 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]:     34:                   td         
Apr 19 16:18:15.137070 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]:   app/views/repositories/show.html.slim:31:in `block in _app_views_repositories_show_html_slim__2245502212515230124_69943756946400'
Apr 19 16:18:15.137070 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]:   app/views/repositories/show.html.slim:30:in `each'
Apr 19 16:18:15.137070 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]:   app/views/repositories/show.html.slim:30:in `_app_views_repositories_show_html_slim__2245502212515230124_69943756946400'
Apr 19 16:18:15.137070 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]:   app/controllers/repositories_controller.rb:17:in `show'
Apr 19 16:18:15.137070 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]: 
Apr 19 16:18:15.137230 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]: 
Apr 19 16:18:15.137230 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]: Processing by ErrorsController#show as HTML
Apr 19 16:18:15.137230 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]:   Parameters: {"id"=>"60", "status"=>"500"}
Apr 19 16:18:15.137832 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]:   Rendered errors/500.html.erb within layouts/errors (0.0ms)
Apr 19 16:18:15.138443 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]:   Rendered errors/_status_title.html.slim (0.0ms)
Apr 19 16:18:15.138443 ops-registry-useast1 docker/nextdoor/portus/portus/71dd60d597eb[4884]: Completed 500 Internal Server Error in 1ms (Views: 0.9ms | ActiveRecord: 0.0ms)
```

**Comment by mssola** [comment]
Argh :sweat:. I've just pushed a fix for that: 9a76f0f1af1a350e2abb4afe2df0a68975138cbc
